### PR TITLE
deploy: update CSI sidecars to latest versions available

### LINKED ISSUE #3867: Update csi sidecars to the latest available versions
# Describe the bug #

Update csi sidecars to the latest available versions.

### PR DESCRIPTION
**Pull request by iPraveenParihar**
Below sidecars are updated with latest available versions

csi-node-driver-registrar: v2.8.0
csi-attacher: v4.3.0
csi-resizer: v1.8.0
csi-snapshotter: v6.2.2
csi-provisioner: v3.5.0

# Describe what this PR does #

Update csi sidecars to the latest available versions 
Fixes #3867 
CC @Rakshith-R

### CODE REVIEW

File: build.env [hunk 1]
```diff
@@ -27,7 +27,7 @@ GOLANGCI_VERSION=v1.47.3
 
 # external snapshotter version
 # Refer: https://github.com/kubernetes-csi/external-snapshotter/releases
-SNAPSHOT_VERSION=v6.1.0
+SNAPSHOT_VERSION=v6.2.2
```
Comment (Madhu-1) [inline build.env]: This is already updated here, this needs to rebase on https://github.com/ceph/ceph-csi/pull/3868 is merged.

**Review by Madhu-1 — DISMISSED** [review]
LGTM

**Review by riya-singhal31 — DISMISSED** [review]
LGTM

**Review by nixpanic — APPROVED** [review]

**Review by riya-singhal31 — APPROVED** [review]
LGTM

**Comment by Rakshith-R** [comment]
@mergifyio queue

**Comment by Rakshith-R** [comment]
@mergifyio requeue

**Comment by nixpanic** [comment]
@mergifyio queue

**Comment by nixpanic** [comment]
@mergifyio rebase

**Pull request merged**